Makefile

```diff
@@ -93,13 +93,13 @@ ci-check: check
 # Note: Go has built-in build caching, so always run `go build`.
 # It will do a better job than using source dependencies to decide if we need to build.
 
-bin/functional-benchmarker: force
```

Comment:
If you remove `force` the binary will never be rebuilt.
If file exists and has no dependencies - make will ignore it.
The purpse of `force` is to force make to run the `go` command and let go figure out the dependencies - if you want to avoid that at the make level, you need to add all the dependencies for the binary to the rule e.g. `$(shell find somewhere/*.go)`

Reply:
An alternative to `force` is `.PHONY`. I use `force` is by force of habit. .PHONY is a GNU extension, the `force` trick works for any make variant, e.g. Solaris or HP-UX . Yes, I am that old.

Reply:
I got the purpose of `force` and wanted to add `.PHONY` instead (but then forgot about that). I think we can rely on the GNU extension to be there, as we're using this for all the other targets.